From 7617a0e6fe2bb7afab6ffda49fc8cd9eca2f035b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Emilio=20Cobos=20=C3=81lvarez?= <emilio@crisal.io>
Date: Thu, 12 Sep 2019 21:09:27 +0000
Subject: [PATCH] Bug 1580963 - Cherry-pick some servo changes from a rustc
 upgrade.

This cherry-picks from Servo commit 98e4a53b724.

Differential Revision: https://phabricator.services.mozilla.com/D45738

--HG--
extra : moz-landing-system : lando
---
 .../style/stylesheets/viewport_rule.rs        | 23 +++++++++++++------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/servo/components/style/stylesheets/viewport_rule.rs b/servo/components/style/stylesheets/viewport_rule.rs
index 8cc858f7a7f69..7d0881e3336a9 100644
--- servo/components/style/stylesheets/viewport_rule.rs
+++ servo/components/style/stylesheets/viewport_rule.rs
@@ -291,15 +291,18 @@ impl<'a, 'b, 'i> DeclarationParser<'i> for ViewportRuleParser<'a, 'b> {
     ) -> Result<Vec<ViewportDescriptorDeclaration>, ParseError<'i>> {
         macro_rules! declaration {
             ($declaration:ident($parse:expr)) => {
-                declaration!($declaration(value: try!($parse(input)),
-                                          important: input.try(parse_important).is_ok()))
+                declaration!($declaration {
+                    value: $parse(input)?,
+                    important: input.try(parse_important).is_ok(),
+                })
             };
-            ($declaration:ident(value: $value:expr, important: $important:expr)) => {
+            ($declaration:ident { value: $value:expr, important: $important:expr, }) => {
                 ViewportDescriptorDeclaration::new(
                     self.context.stylesheet_origin,
                     ViewportDescriptor::$declaration($value),
-                    $important)
-            }
+                    $important,
+                )
+            };
         }
 
         macro_rules! ok {
@@ -311,8 +314,14 @@ impl<'a, 'b, 'i> DeclarationParser<'i> for ViewportRuleParser<'a, 'b> {
                 let important = input.try(parse_important).is_ok();
 
                 Ok(vec![
-                    declaration!($min(value: shorthand.0, important: important)),
-                    declaration!($max(value: shorthand.1, important: important)),
+                    declaration!($min {
+                        value: shorthand.0,
+                        important: important,
+                    }),
+                    declaration!($max {
+                        value: shorthand.1,
+                        important: important,
+                    }),
                 ])
             }};
         }
